wallet/wallet_v4.tact

```diff
@@ -0,0 +1,121 @@
+interface Plugin {
+    internal requested_funds#f06c7567(query_id: Uint64);
+    internal notify_removed#e4737472(query_id: Uint64);
+    internal notify_installed#6e6f7465(query_id: Uint64);
+    internal notify_removed#64737472(query_id: Uint64);
+}
+
+contract Wallet {
+    var seqno: Uint32
+    var subwallet: Uint32
+    var pubkey: Pubkey
+    var plugins: HashSet[Plugin]
+
+    bounced default() {
+        // ignore bounced messages
+    }
+
+    internal default() {
+        // ignore simple transfers
+    }
+
+    internal request_funds#706c7567(
+        query_id: Uint64,
+        requested_amount: Coins,
+        requested_extra: ExtraCurrencies,
+        msg.sender: Plugin,
+    ) {
+        verify(self.balance() - msg.amount >= requested_amount, "Not enough funds");
+
+        send(InboundCoins)
+        msg.sender.requested_funds(
+            query_id,
+            msg.amount: requested_amount,
+            msg.extra: requested_extra,
+        );
+    }
+
+    internal remove_plugin#64737472(query_id: Uint64, msg.sender: Plugin) {
+        self.remove_plugin_from_list(msg.sender);
+
+        if msg.bounceable {
+            send(InboundCoins) msg.sender.notify_removed(query_id);
+        }
+    }
+
+    external pre_condition(body: Slice) -> Slice {
+        let sign = body.read(Signature);
+        if sign.verify(body, self.pubkey) {
+            self.accept();
+            self.seqno += 1;
+            self.commit();
+            return body;
+        } else {
+            thrown("Invalid signature");
+        }
+    }
+
+    external simple_transfer#0(messages: List[(Cell, Sendmode)]) -> Slice {
+        for (msg, flags) in messages {
+            send_raw_msg(msg, flags);
+        }
+    }
+
+    external destroy_and_install_plugin#1(
+        workchain: Int8,
+        balance: Coins,
+        state_init: Ref[StateInit],
+        body: Cell,
```

Comment:
Maybe for consistency we should not do `Cell=Ref[Any]`, but instead write `Ref[Cell]` when we have an untyped cell. Or maybe even Ref[Cell[Any]], where Ref is defined with a default: `type Ref[T=Cell[Any]]`? Not sure what's the best way forward here.

I feel like we want to have one word to indicate refs in the type definition. So maybe we could unify Ref/Cell/List as `Ref` + `RefList` and use `Cell` as opaque type?

Or, the same idea but the names flipped:

* Cell[T] for refs.
* Cells[T,B] or CellList for lists. But CellList spells badly with triple "L".

Gotta think about this more.

Reply:
`Ref` is a typed reference, while `Cell` is exactly what it means - it is a TVM cell. `Ref` is top-level type, while `Cell` is a raw type. I think we shouldn't rename `Cell` type.

I'm not sure we want to introduce `Any` type which mean nothing outside of `Ref` type.